airflow/ui/src/pages/Variables/AddVariableForm.tsx

```diff
@@ -0,0 +1,131 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Box, Field, HStack, Input, Spacer, Textarea } from "@chakra-ui/react";
+import { useEffect } from "react";
+import { Controller, useForm } from "react-hook-form";
+import { FiSave } from "react-icons/fi";
+
+import { ErrorAlert } from "src/components/ErrorAlert";
+import { Button } from "src/components/ui";
+import { useAddVariable } from "src/queries/useAddVariable";
+
+export type AddVariableBody = {
+  description: string | undefined;
+  key: string;
+  value: string;
+};
+
+type AddVariableFormProps = {
+  readonly onClose: () => void;
+};
+
+const AddVariableForm = ({ onClose }: AddVariableFormProps) => {
+  const { addVariable, error, isPending } = useAddVariable(onClose);
+
+  const {
+    control,
+    formState: { isDirty },
+    handleSubmit,
+    reset,
+    watch,
+  } = useForm<AddVariableBody>({
+    defaultValues: {
+      description: "",
+      key: "",
+      value: "",
+    },
+  });
+
+  const { key, value } = watch();
+
+  useEffect(() => {
+    reset({
+      description: "",
+      key: "",
+      value: "",
+    });
+  }, [reset]);
+
+  const onSubmit = (data: AddVariableBody) => {
+    addVariable(data);
+  };
+
+  return (
+    <>
+      <Controller
+        control={control}
+        name="key"
+        render={({ field }) => (
+          <Field.Root required>
+            <Field.Label fontSize="md">
+              Key <Field.RequiredIndicator />
+            </Field.Label>
+            <Input {...field} required size="sm" />
```

Comment:
Can you set a `maxLength` preventing the user to enter too much text and then having a response from server as error?

![image](https://github.com/user-attachments/assets/f2a0ea51-6f8d-4531-9889-64ff52f7b663)
```suggestion
            <Input {...field} required size="sm" maxLength={250} />
```